### PR TITLE
linuxPackages.rtl8188eus-aircrack: unstable-2021-05-04 -> unstable-2022-03-19

### PR DESCRIPTION
**Pull request by AtilaSaraiva**
###### Description of changes

Not exactly sure what changed.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AtilaSaraiva** [comment]
I wasn't able to build everything on my machine, my RAM was depleted, I hope ofborg can do a better job.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 187322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>37 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.hardkernel_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_lqx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_zen.rtl8188eus-aircrack</li>
    <li>linuxPackages-libre.rtl8188eus-aircrack</li>
    <li>linuxPackages.rtl8188eus-aircrack</li>
    <li>linuxPackages_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxPackages_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxPackages_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxPackages_5_15_hardened.rtl8188eus-aircrack</li>
    <li>linuxPackages_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxPackages_5_4_hardened.rtl8188eus-aircrack</li>
    <li>linuxPackages_hardened.rtl8188eus-aircrack</li>
    <li>linuxPackages_hardkernel_latest.rtl8188eus-aircrack</li>
    <li>linuxPackages_latest-libre.rtl8188eus-aircrack</li>
    <li>linuxPackages_latest.rtl8188eus-aircrack</li>
    <li>linuxPackages_lqx.rtl8188eus-aircrack</li>
    <li>linuxPackages_testing_bcachefs.rtl8188eus-aircrack</li>
    <li>linuxPackages_xanmod.rtl8188eus-aircrack</li>
    <li>linuxPackages_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxPackages_xanmod_tt.rtl8188eus-aircrack</li>
    <li>linuxPackages_zen.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_9.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4.rtl8188eus-aircrack</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Note the notation:

`broken = kernel.kernelAtLeast "5.15" || kernel.isHardened;`

Is it what you had in mind?

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 187322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_18.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_lqx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_zen.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_9.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_hardened.rtl8188eus-aircrack (linuxKernel.packages.linux_5_15_hardened.rtl8188eus-aircrack)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8188eus-aircrack</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Try this patch:
https://github.com/aircrack-ng/rtl8188eus/pull/193

**Comment by AtilaSaraiva** [comment]
Done, could you please test again?

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 187322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>21 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_9.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_hardened.rtl8188eus-aircrack (linuxKernel.packages.linux_5_15_hardened.rtl8188eus-aircrack)</li>
    <li>linuxKernel.packages.linux_5_18.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_lqx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_zen.rtl8188eus-aircrack</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 187322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_18.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_lqx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_zen.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_9.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_hardened.rtl8188eus-aircrack (linuxKernel.packages.linux_5_15_hardened.rtl8188eus-aircrack)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8188eus-aircrack</li>
  </ul>
</details>

File: pkgs/os-specific/linux/rtl8188eus-aircrack/default.nix [hunk 1]
```diff
@@ -25,17 +25,22 @@ stdenv.mkDerivation {
...
     mkdir -p "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
   '';
 
-  enableParallelBuilding = true;
-
   meta = with lib; {
     description = "RealTek RTL8188eus WiFi driver with monitor mode & frame injection support";
     homepage = "https://github.com/aircrack-ng/rtl8188eus";
     license = licenses.gpl2Only;
     maintainers = with maintainers; [ fortuneteller2k ];
```
Comment (superherointj) [inline pkgs/os-specific/linux/rtl8188eus-aircrack/default.nix:44]:
```suggestion
    maintainers = with maintainers; [ fortuneteller2k ];
    broken = (lib.versionAtLeast kernel.version "5.17") || (kernel.version == "5.4" && kernel.isHardened);
```

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 187322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_18.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_lqx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_zen.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_9.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_hardened.rtl8188eus-aircrack (linuxKernel.packages.linux_5_15_hardened.rtl8188eus-aircrack)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8188eus-aircrack</li>
  </ul>
</details>

File: pkgs/os-specific/linux/rtl8188eus-aircrack/default.nix [hunk 2]
```diff
@@ -25,17 +25,23 @@ stdenv.mkDerivation {
...
-    broken = kernel.kernelAtLeast "5.15" || kernel.isHardened;
+    broken = (lib.versionAtLeast kernel.version "5.17") || (kernel.version == "5.4" && kernel.isHardened);
```
Comment (superherointj) [inline pkgs/os-specific/linux/rtl8188eus-aircrack/default.nix:45]:
```suggestion
    broken = (lib.versionAtLeast kernel.version "5.17") || ((lib.versions.majorMinor kernel.version) == "5.4" && kernel.isHardened);
```

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 187322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_18.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_lqx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_zen.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_9.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_hardened.rtl8188eus-aircrack (linuxKernel.packages.linux_5_15_hardened.rtl8188eus-aircrack)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8188eus-aircrack</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 187322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_18.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_lqx.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_zen.rtl8188eus-aircrack</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_4_9.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_5_15.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_hardened.rtl8188eus-aircrack (linuxKernel.packages.linux_5_15_hardened.rtl8188eus-aircrack)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_libre.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8188eus-aircrack</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**